pkg/humio/client.go

```diff
@@ -246,18 +241,23 @@ func (c *Client) Stream(method string, path string, query Query, ch chan Streami
 		return err
 	}
 	defer func() {
-		if err := res.Body.Close(); err != nil {
-			log.DefaultLogger.Warn("Failed to close response body", "error", err)
-		}
+		res.Body.Close()
 	}()
 
 	d := json.NewDecoder(res.Body)
 
 	for {
+		select {
+		case <-ctx.Done():
+			if ctx.Err() == context.Canceled {
+				return nil
+			}
+			return ctx.Err()
+		default:
+		}
 		var result StreamingResults
```

Comment:
Should everything here sit in the `default` block?

Reply:
It's a golang style choice. such fashion